code/cmdline/cmdline.cpp

```diff
@@ -1036,6 +1036,105 @@ char *cmdline_parm::str()
 	return args;
 }
 
+#ifdef SCP_UNIX
+// Return a vector with all filesystem names of "parent/dir" relative to parent.
+// dir must not contain a slash.
+static SCP_vector<SCP_string> unix_get_single_dir_names(SCP_string dir, SCP_string parent)
+{
+	DIR *dp;
+	if ((dp = opendir(parent.c_str())) == NULL) {
+		Error(LOCATION, "Can't open directory '%s' -- %d", parent.c_str(), errno);
+	}
+
+	SCP_vector<SCP_string> ret;
+
+	dirent *dirp;
+	while ((dirp = readdir(dp)) != NULL) {
+		if (!stricmp(dirp->d_name, dir.c_str())) {
+			ret.push_back(dirp->d_name);
+		}
+	}
+	(void)closedir(dp);
+
+	return ret;
+}
+
+// Return a vector with all filesystem names of "parent/dir" relative to parent.
+// Recurses to deal with slashes in dir.
+static SCP_vector<SCP_string> unix_get_dir_names(SCP_string dir, SCP_string parent)
+{
+	size_t slash = dir.find_first_of("/\\");
+
+	// no subdirectories, no need to recurse
+	if (slash == std::string::npos) {
+		return unix_get_single_dir_names(dir, parent);
+	}
+
+	// get the names of the first component of dir
+	SCP_vector<SCP_string> this_dir_names = unix_get_single_dir_names(dir.substr(0, slash), parent);
+
+	SCP_string rest = dir.substr(slash + 1);
+
+	SCP_vector<SCP_string> ret;
+
+	// search for the rest of dir in each of these
+	SCP_vector<SCP_string>::iterator ii, end = this_dir_names.end();
+	for (ii = this_dir_names.begin(); ii != end; ++ii) {
+		SCP_string this_dir_path = parent + "/" + *ii;
+		SCP_vector<SCP_string> mod_path = unix_get_dir_names(rest, this_dir_path);
+
+		// add all found paths relative to parent
+		SCP_vector<SCP_string>::iterator ii2, end2 = mod_path.end();
+		for (ii2 = mod_path.begin(); ii2 != end2; ++ii2) {
+			ret.push_back(*ii + "/" + *ii2);
+		}
+	}
+
+	return ret;
+}
+
+// For case sensitive filesystems (e.g. Linux/BSD) perform case-insensitive dir matches.
+static void handle_unix_modlist(char **modlist, int *len)
+{
+	char cur_dir[CF_MAX_PATHNAME_LENGTH];
+
+	if ( !_getcwd(cur_dir, CF_MAX_PATHNAME_LENGTH ) ) {
+		Error(LOCATION, "Can't get current working directory -- %d", errno );
+	}
+
+	// search filesystem for given paths
+	SCP_vector<SCP_string> mod_paths;
+	for (char *cur_mod = strtok(*modlist, ","); cur_mod != NULL; cur_mod = strtok(NULL, ","))
+	{
+		SCP_vector<SCP_string> this_mod_paths = unix_get_dir_names(cur_mod, cur_dir);
+		if (this_mod_paths.empty()) {
+			ReleaseWarning(LOCATION, "Can't find mod '%s'", cur_mod);
+		}
+		mod_paths.insert(mod_paths.end(), this_mod_paths.begin(), this_mod_paths.end());
+	}
+
+	// create new char[] to replace modlist
+	size_t total_len = 0;
+	SCP_vector<SCP_string>::iterator ii, end = mod_paths.end();
+	for (ii = mod_paths.begin(); ii != end; ++ii) {
+		total_len += (*ii).length() + 1;
```

Comment:
You can access a member of the object an iterator points to with `->` (in this case `ii->length()` will work).